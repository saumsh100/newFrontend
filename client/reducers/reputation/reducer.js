import { fromJS } from 'immutable';

import {
  FETCH_REPUTATION,
  FETCH_REPUTATION_SUCCESS
} from './constants';

const initialState = fromJS({
  status: 'loading', // set only to 'loading', 'success' or 'error'
  lastFetched: null,
  data: {
    sourcesNotFound: null,
    sourcesFound: null,
    listingScore: null,
    listingPointScore: {
      industryAverage: null,
      industryLeadersAverage: null,
      pointScore: null
    },
    citationsFound: null,
    sourcesFoundWithErrors: null,
    accuracyScore: null
  }
})

function reputationReducer(state = initialState, action) {
  switch (action.type) {
    case FETCH_REPUTATION:
      return state.set('status', 'loading')
    case FETCH_REPUTATION_SUCCESS:
      return state
        .set('status', 'success')
        .set('lastFetched', (new Date()).toLocaleString())
        .set('data', action.data)
    default:
      return state
  }
}

export default reputationReducer;