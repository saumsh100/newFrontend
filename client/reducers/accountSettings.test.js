
import reducer, { initialState, setServiceId, setPractitionerId } from './accountSettings';

const setServiceIdAction = setServiceId({ id: '123123123' });
const setPractitionerIdAction = setPractitionerId({ id: '33333' });

describe('accountSettings reducer', () => {
  test('action creator works', () => {
    expect(typeof setServiceId).toBe('function');
    expect(typeof setPractitionerId).toBe('function');
  });

  test('setServiceId', () => {
    expect(setServiceIdAction).toMatchSnapshot();
  });

  test('setPractitionerId', () => {
    expect(setPractitionerIdAction).toMatchSnapshot();
  });

  test('reducer work', () => {
    expect(reducer(initialState, {})).toEqual(initialState);
    expect(reducer(initialState, setServiceIdAction)).toMatchSnapshot();
    expect(reducer(initialState, setPractitionerIdAction)).toMatchSnapshot();
  });
});
