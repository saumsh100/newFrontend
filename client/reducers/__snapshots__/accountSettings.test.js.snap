// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accountSettings reducer reducer work 1`] = `
Immutable.Map {
  "serviceId": "123123123",
  "practitionerId": null,
}
`;

exports[`accountSettings reducer reducer work 2`] = `
Immutable.Map {
  "serviceId": null,
  "practitionerId": "33333",
}
`;

exports[`accountSettings reducer setPractitionerId 1`] = `
Object {
  "payload": Object {
    "id": "33333",
  },
  "type": "@accountSettings/SET_PRACTITIONER_ID",
}
`;

exports[`accountSettings reducer setServiceId 1`] = `
Object {
  "payload": Object {
    "id": "123123123",
  },
  "type": "@accountSettings/SET_SERVICE_ID",
}
`;
